datasets/datasets_decorator.py

```diff
@@ -27,6 +29,23 @@ def step_wrapper(*args, **kwargs):
                 _snake_name = _pascal_to_snake_case(dataset.name)
                 setattr(self, _snake_name, dataset)
 
+            # Transformation integration
+            if hasattr(self, "name"):
+                flow_name = self.name
+            else:
+                flow_name = type(self).__name__
+
+            add_txf_attributes(self)
+            if not self._txf_registered_datasets.get(flow_name, None):
+                self._txf_registered_datasets[flow_name] = [dataset]
+            else:
+                self._txf_registered_datasets[flow_name].append(dataset)
+
+            if len(self._txf_callbacks):
+                for _, callback in self._txf_callbacks.items():
+                    callback()
```

Comment:
Is this callback actually calling the transform?  better named `txf_callback`.
It'll help to see an example transform in unit tests and/or tutorials.